File: src/android-app/app/src/main/java/com/example/mobileapp/services/ApiService.java
package com.example.mobileapp.services;

import com.example.mobileapp.models.*;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.*;

public interface ApiService {
    
    @POST("/api/mobile/login")
    Call<LoginResponse> login(@Body LoginRequest request);
    
    @GET("/api/mobile/products")
    Call<ProductListResponse> getProducts();
    
    @GET("/api/mobile/products/search")
    Call<ProductListResponse> searchProducts(@Query("keyword") String keyword);
    
    @GET("/api/mobile/products/{id}")
    Call<ProductDetailResponse> getProductDetail(@Path("id") int productId);
    
    @POST("/api/mobile/purchases")
    Call<PurchaseResponse> purchaseProduct(@Body PurchaseRequest request);
    
    @POST("/api/mobile/favorites")
    Call<FavoriteResponse> addFavorite(@Body FavoriteRequest request);
    
    @DELETE("/api/mobile/favorites/{id}")
    Call<ResponseBody> removeFavorite(@Path("id") int favoriteId);
    
    @GET("/api/mobile/favorites")
    Call<FavoriteListResponse> getFavorites();
}

File: android-app/app/src/main/java/com/example/mobileapp/services/ApiService.kt
package com.example.mobileapp.services

import com.example.mobileapp.models.*
import okhttp3.ResponseBody
import retrofit2.Response
import retrofit2.http.*

interface ApiService {
    
    @POST("/api/mobile/login")
    suspend fun login(@Body request: LoginRequest): Response<LoginResponse>
    
    @GET("/api/mobile/products")
    suspend fun getProducts(): Response<ProductListResponse>
    
    @GET("/api/mobile/products/search")
    suspend fun searchProducts(@Query("keyword") keyword: String): Response<ProductListResponse>
    
    @GET("/api/mobile/products/{id}")
    suspend fun getProductDetail(@Path("id") productId: Int): Response<ProductDetailResponse>
    
    @POST("/api/mobile/purchases")
    suspend fun purchaseProduct(@Body request: PurchaseRequest): Response<PurchaseResponse>
    
    @POST("/api/mobile/favorites")
    suspend fun addFavorite(@Body request: FavoriteRequest): Response<FavoriteResponse>
    
    @DELETE("/api/mobile/favorites/{id}")
    suspend fun removeFavorite(@Path("id") favoriteId: Int): Response<ResponseBody>
    
    @GET("/api/mobile/favorites")
    suspend fun getFavorites(): Response<FavoriteListResponse>
}
